pkg/querier/queryrange/queryrange.proto

```diff
@@ -20,9 +20,10 @@ message PrometheusRequest {
   google.protobuf.Duration timeout = 5 [(gogoproto.stdduration) = true, (gogoproto.nullable) = false];
   string query = 6;
   CachingOptions cachingOptions = 7 [(gogoproto.nullable) = false];
+  repeated PrometheusHeader Headers = 8 [(gogoproto.jsontag) = "-"];
 }
 
-message PrometheusResponseHeader {
+message PrometheusHeader {
```

Comment:
I'm not sure it makes a lot of difference, but I think I would be happier with adding `PrometheusRequestHeader` as a duplicate rather than changing the response proto.